Fix TypeScript build errors: GardenScene.init() signature mismatch

## Problem
The deploy workflow fails because `npx tsc --noEmit` produces errors:

```
src/main.ts(22,42): error TS2345: Argument of type 'GardenScene' is not assignable to parameter of type 'Scene'.
src/scenes/GardenScene.ts(14,9): error TS2416: Property 'init' in type 'GardenScene' is not assignable to the same property in base type 'Scene'.
```

## Root Cause
GardenScene.init() receives Application but the Scene interface expects SceneContext. The SceneContext pattern was introduced but GardenScene was not updated to match.

## Fix
Update GardenScene.init() to accept SceneContext instead of Application, and adjust internal references accordingly.

## Impact
- All 5 deploy workflow runs have failed
- GitHub Pages for Flora has never successfully deployed

## Acceptance Criteria
- npx tsc --noEmit passes with zero errors
- Deploy workflow succeeds and GitHub Pages is live

--- a/src/scenes/GardenScene.ts
+++ b/src/scenes/GardenScene.ts
@@ -1,5 +1,5 @@
-import { Application, Container, Text } from 'pixi.js';
-import type { Scene } from '../core';
+import { Container, Text } from 'pixi.js';
+import type { Scene, SceneContext } from '../core';
 import { GardenGrid } from '../entities/GardenGrid';
 import { TileState } from '../entities/Tile';
 import { GridSystem } from '../systems/GridSystem';
@@ -11,9 +11,8 @@ export class GardenScene implements Scene {
   private gridSystem!: GridSystem;
   private infoText!: Text;
 
-  async init(app: Application): Promise<void> {
-    const sceneManager = app.stage.children[0] as Container;
-    sceneManager.addChild(this.container);
+  async init(ctx: SceneContext): Promise<void> {
+    ctx.sceneManager.stage.addChild(this.container);
 
     // Initialize garden grid (8x8)
     this.grid = new GardenGrid({
@@ -25,7 +24,7 @@ export class GardenScene implements Scene {
 
     // Initialize grid rendering system
     this.gridSystem = new GridSystem(this.grid);
-    this.gridSystem.centerInViewport(app.screen.width, app.screen.height);
+    this.gridSystem.centerInViewport(ctx.app.screen.width, ctx.app.screen.height);
     this.container.addChild(this.gridSystem.getContainer());
 
     // Add some demo state to tiles
@@ -64,7 +63,7 @@ export class GardenScene implements Scene {
       },
     });
     this.infoText.anchor.set(0.5, 0);
-    this.infoText.x = app.screen.width / 2;
+    this.infoText.x = ctx.app.screen.width / 2;
     this.infoText.y = 20;
     this.container.addChild(this.infoText);
 
@@ -72,7 +71,7 @@ export class GardenScene implements Scene {
     this.gridSystem.update();
   }
 
-  update(_delta: number): void {
+  update(_delta: number, _ctx: SceneContext): void {
     // Update grid system (re-renders if state changed)
     this.gridSystem.update();
 

